src/historywork/DownloadBucketsWork.cpp

```diff
@@ -17,16 +15,23 @@ DownloadBucketsWork::DownloadBucketsWork(
     Application& app, WorkParent& parent,
     std::map<std::string, std::shared_ptr<Bucket>>& buckets,
     std::vector<std::string> hashes, TmpDir const& downloadDir)
-    : Work{app, parent, "download-and-verify-buckets"}
+    : BatchWork{app, parent, "download-verify-buckets"}
     , mBuckets{buckets}
-    , mHashes{std::move(hashes)}
+    , mHashes{hashes}
+    , mNextBucketIndex{hashes.begin()}
```

Comment:
1. you forgot to rename `mNextBucketIndex` into ``mNextBucketIterator``
2. this is wrong: `mNextBucketIterator` should be `mHashes.begin()` not `hashes.begin()`